docs/NNNN-hash-and-id-n-tuple-storage-layout.md

```diff
@@ -0,0 +1,323 @@
+| ..hor/rib:le-$id | 08319766fb6c2935dd175b94267717e0 | `08/31/97/66/fb/6c/29/35/dd/17/5b/94/26/77/17/%2e%2ehor%2frib%3ale-%24id` |
+
+#### Storage Hierarchy
+
+```
+[storage_root]/
+├── 0=ocfl_1.0
+├── ocfl_layout.json
+├── NNNN-hashed-n-tuple-id-layout.json
+├── 08/
+│   └── 31/
+│       └── 97/
+│           └── 66/
+│               └── fb/
+│                   └── 6c/
+│                       └── 29/
+│                           └── 35/
+│                               └── dd/
+│                                   └── 17/
+│                                       └── 5b/
+│                                           └── 94/
+│                                               └── 26/
+│                                                   └── 77/
+│                                                       └── 17/
+│                                                           └── %2e%2ehor%2frib%3ale-%24id/
+│                                                               ├── 0=ocfl_object_1.0
+│                                                               ├── inventory.json
+│                                                               ├── inventory.json.sha512
+│                                                               └── v1 [...]
+└── ff/
+    └── 75/
+        └── 53/
+            └── 44/
+                └── 92/
+                    └── 48/
+                        └── 5e/
+                            └── ab/
+                                └── b3/
+                                    └── 9f/
+                                        └── 86/
+                                            └── 35/
+                                                └── 67/
+                                                    └── 28/
+                                                        └── 88/
+                                                            └── object-01/
+                                                                ├── 0=ocfl_object_1.0
+                                                                ├── inventory.json
+                                                                ├── inventory.json.sha512
+                                                                └── v1 [...]
+```
+
+### Example 3
+
+This example demonstrates what happens when `tupleSize` and `numberOfTuples` are set to `0`. This is an edge case and not a recommended configuration.
+
+#### Parameters
+
+```json
+{
+    "digestAlgorithm": "sha256",
+    "tupleSize": 0,
+    "numberOfTuples": 0
+}
+```
+
+#### Mappings
+
+| Object ID | Digest | Object Root Path |
+| --- | --- | --- |
+| object-01 | 3c0ff4240c1e116dba14c7627f2319b58aa3d77606d0d90dfc6161608ac987d4 | `object-id` |
+| ..hor/rib:le-$id | 487326d8c2a3c0b885e23da1469b4d6671fd4e76978924b4443e9e3c316cda6d | `%2e%2ehor%2frib%3ale-%24id` |
+
+#### Storage Hierarchy
+
+```
+[storage_root]/
+├── 0=ocfl_1.0
+├── ocfl_layout.json
+├── NNNN-hashed-n-tuple-id-layout.json
```

Comment:
This file should now be at `extensions/hash-and-id-n-tuple-storage-layout/config.json`